src/isomorphic/modern/class/ReactBaseClasses.js

```diff
@@ -144,7 +144,27 @@ ReactPureComponent.prototype.constructor = ReactPureComponent;
 Object.assign(ReactPureComponent.prototype, ReactComponent.prototype);
 ReactPureComponent.prototype.isPureReactComponent = true;
 
+function ReactAsyncComponent(props, context, updater) {
+  // Duplicated from ReactComponent.
+  this.props = props;
+  this.context = context;
+  this.refs = emptyObject;
+  // We initialize the default updater but the real one gets injected by the
+  // renderer.
+  this.updater = updater || ReactNoopUpdateQueue;
+}
+
+ReactAsyncComponent.prototype = new ComponentDummy();
```

Comment:
Might be worth adding a variable here. `var ReactAsyncComponentPrototype = ` here so that you can use that instead of rereading the prototype below. Save a few bytes.